src/rdkafka_telemetry_encode.c

```diff
@@ -183,105 +259,180 @@ void *rd_kafka_telemetry_encode_metrics(rd_kafka_t *rk, size_t *size) {
+                        data_points[i]->value.as_double =
+                            METRICS[metrics_to_encode[i]]
+                                .calculate_value(rk)
+                                .doubleValue;
+                }
+
+                data_points[i]->time_unix_nano = rd_clock();
+
+                //    TODO: Add data point attributes as needed
+                //    opentelemetry_proto_common_v1_KeyValue attribute =
+                //    opentelemetry_proto_common_v1_KeyValue_init_zero;
+                //    attribute.key.funcs.encode = &encode_string;
+                //    attribute.key.arg = "type";
+                //    attribute.has_value = true;
+                //    attribute.value.which_value =
+                //    opentelemetry_proto_common_v1_AnyValue_string_value_tag;
+                //    attribute.value.value.string_value.funcs.encode =
+                //    &encode_string; attribute.value.value.string_value.arg =
+                //    "heap";
+                //
+                //    data_point.attributes.funcs.encode = &encode_key_value;
+                //    data_point.attributes.arg = &attribute;
+
+
+                switch (METRICS[metrics_to_encode[i]].type) {
+
+                case METRIC_TYPE_SUM: {
+                        sum = rd_malloc(
+                            sizeof(opentelemetry_proto_metrics_v1_Sum));
+                        memset(sum, 0,
+                               sizeof(opentelemetry_proto_metrics_v1_Sum));
+                        sum->data_points.funcs.encode =
+                            &encode_number_data_point;
+                        sum->data_points.arg = data_points[i];
+                        sum->aggregation_temporality =
+                            opentelemetry_proto_metrics_v1_AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE;
+                        sum->is_monotonic = true;
+                        metrics[i]->which_data =
+                            opentelemetry_proto_metrics_v1_Metric_sum_tag;
+                        metrics[i]->data.sum = *sum;
+                        rd_free(sum);
+                        break;
+                }
+                case METRIC_TYPE_GAUGE: {
+                        gauge = rd_malloc(
+                            sizeof(opentelemetry_proto_metrics_v1_Gauge));
+                        memset(gauge, 0,
+                               sizeof(opentelemetry_proto_metrics_v1_Gauge));
+                        gauge->data_points.funcs.encode =
+                            &encode_number_data_point;
+                        gauge->data_points.arg = data_points[i];
+                        metrics[i]->which_data =
+                            opentelemetry_proto_metrics_v1_Metric_gauge_tag;
+                        metrics[i]->data.gauge = *gauge;
+                        rd_free(gauge);
+                        break;
+                }
+                default:
+                        break;
```

Comment:
`rd_assert(!*"<some error message saying this should be impossible>");`

Reply:
Done